pkg/controller/migmigration/task.go

```diff
@@ -520,6 +521,21 @@ func (t *Task) Run() error {
 		if restore == nil {
 			return errors.New("Restore not found")
 		}
+		report, err := t.ensureDestinationStagePodsStarted()
+		if err != nil {
+			return liberr.Wrap(err)
+		}
+		if report.failed {
+			t.fail(DestinationStagePodsFailed, report.reasons)
+			break
+		}
+		if report.started {
```

Comment:
@alaypatel07 Usage of `t.next()` is not appropriate after ensure `DestinationStagePodsStarted()` call. It should be only called once per Phase to increment the Phase counter by 1.